src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -520,6 +520,7 @@ async Task<bool> NextResult(bool async, bool isConsuming = false, CancellationTo
                 switch (msg.Code)
                 {
                 case BackendMessageCode.DataRow:
+                    Connector.State = ConnectorState.Fetching;
```

Comment:
Is this a bug fix?

Reply:
Nah, just an optimization (currently we attempt to set it every time we handle DataRow). To me makes sense to at least set only once per NextResult call.

Reply:
The end result should be the same, just one less call per Read.